src/main/java/com/buyanywhere/productcatalog/controllers/v1/CategoryController.java

```diff
@@ -1 +1,40 @@
+package com.buyanywhere.productcatalog.controllers.v1;
 
+import com.buyanywhere.productcatalog.exceptions.CategoryNotFoundException;
+import com.buyanywhere.productcatalog.exceptions.InvalidCategoryException;
+import org.springframework.web.bind.annotation.*;
+import com.buyanywhere.productcatalog.models.Category;
+import com.buyanywhere.productcatalog.repositories.CategoryRepository;
+
+@RestController
+@RequestMapping("/v1/category")
+
+public class CategoryController {
+
+    private CategoryRepository repository;
+
+    public CategoryController(CategoryRepository repository) {
+        this.repository = repository;
+    }
+
+    @GetMapping(value = "/{id}")
+    public Category get(@PathVariable("id") long id) throws CategoryNotFoundException {
+        if (!repository.findById(id).isPresent() || repository.findById(id).get().isDeleted()) {
+            throw new CategoryNotFoundException(id);
+        }
+        return repository.findById(id).get();
+    }
+
+    @PostMapping
+    public Category post(@RequestBody Category category) {
+        return repository.save(category);
+    }
+
+    @PutMapping
+    public Category put(@RequestBody Category category) throws CategoryNotFoundException {
+        if (!category.isValid()) {
+            throw new InvalidCategoryException("Invalid data: Name: " + category.getName() + ", Order: " + category.getDisplayOrder());
```

Comment:
I will accept this, but once every one's code is in master please review it for correct usage

Reply:
Sure @maximilianotopp 
I will  refactor this method in order to remove the getName() and getDisplayOrder() method.